pkg/kubelet/cri/remote/remote_runtime.go

```diff
@@ -1204,3 +1205,36 @@ func (r *remoteRuntimeService) CheckpointContainer(options *runtimeapi.Checkpoin
 
 	return nil
 }
+
+func (r *remoteRuntimeService) GetContainerEvents(containerEventsCh chan *runtimeapi.ContainerEventResponse) error {
+	errCh := make(chan error)
+
+	if r.useV1API() {
+		containerEventsStreamingClient, err := r.runtimeClient.GetContainerEvents(context.Background(), &runtimeapi.GetEventsRequest{})
+		if err != nil {
+			klog.ErrorS(err, "GetContainerEvents failed to get streaming client")
+			return err
+		}
+
+		go func() {
+			for {
+				resp, err := containerEventsStreamingClient.Recv()
+				if err == io.EOF {
+					errCh <- err
+					return
+				}
+				if err != nil {
+					klog.ErrorS(err, "cannot receive container event")
+					errCh <- err
+					return
+				}
+				if resp != nil {
+					containerEventsCh <- resp
+					klog.V(4).InfoS("container event received", "resp", resp)
+				}
+			}
+		}()
+		return <-errCh
```

Comment:
another nit: we seem to call GetContainerEvents below in a new goroutine (and ignore the error) whereas we also spawn a new routine here. that seems odd to me, but I don't have the context from testing to know if it's intentional or required.